features/vat-replacement/run.js

```diff
@@ -0,0 +1,239 @@
+            'There seems to be a special case.' + '\n'
+        errMsg += await buildTaxRateDraftJson(invalidGermanTaxRateList, taxRateIdToTaxCategoryMap)
+        console.error(errMsg)
+    }
+    if (standardVATs.length > 1) {
+        errMsg  = 'We are sorry but there are several tax rates for "DE" with a percentage ' +
+            'of ' + vatConstant.TAX_RATE_STANDARD_OLD*100 + ' percent. Please update the tax rate manually. ' + '\n'
+        errMsg += await buildTaxRateDraftJson(standardVATs, taxRateIdToTaxCategoryMap)
+        console.error(errMsg)
+    }
+    if (reducedVATs.length > 1) {
+        errMsg  = 'We are sorry but there are several tax rates for "DE" with a percentage ' +
+            'of ' + vatConstant.TAX_RATE_REDUCED_OLD*100 + ' percent. Please update the tax rate manually. ' + '\n'
+        errMsg += await buildTaxRateDraftJson(reducedVATs, taxRateIdToTaxCategoryMap)
+        console.error(errMsg)
+    }
+
+    return validGermanTaxRateList
+}
+
+async function buildTaxRateDraftJson(taxRateList, taxRateIdToTaxCategoryMap) {
+    let msg = ''
+    taxRateList.forEach(item => {
+        msg += 'Tax Category Name : '+taxRateIdToTaxCategoryMap.get(item.id).name + '\n'
+        msg += JSON.stringify(item) + '\n'
+    })
+    return msg
+}
+
+async function buildTaxRateIdToTaxCategoryMap(items) {
+    const map = new Map()
+    for (const item of items) {
+        let taxCategoryItem = {
+            id : item.id,
+            version : item.version,
+            name : item.name,
+            key : item.key
+        }
+        item.rates.forEach( rate => map.set(rate.id,  taxCategoryItem))
+    }
+    return map
+}
+
+async function getUpdateJsonObj(taxRateDraft, taxRateIdToTaxCategoryMap ) {
+    const taxCategoryItem = taxRateIdToTaxCategoryMap.get(taxRateDraft.id)
+
+    const updateJsonObj = {
+        version : taxCategoryItem.version,
+        actions : [
+            {
+                action: "replaceTaxRate",
+                taxRateId: taxRateDraft.id,
+                taxRate: {
+                    name: taxRateDraft.name,
+                    amount: taxRateDraft.amount,
+                    includedInPrice : taxRateDraft.includedInPrice,
+                    country: taxRateDraft.country
+                }
+            }
+        ]
+    }
+    return updateJsonObj
+
+}
+
+async function processTaxRate({ctpClient, validGermanTaxRateList, taxRateIdToTaxCategoryMap, taxRateType, isDryRun}) {
+    let oldTaxRate =  0
+    let newTaxRate =  0
+    if (taxRateType === vatConstant.TAX_RATE_TYPE_STANDARD) {
+        oldTaxRate = vatConstant.TAX_RATE_STANDARD_OLD
+        newTaxRate = vatConstant.TAX_RATE_STANDARD_NEW
+    } else {
+        oldTaxRate = vatConstant.TAX_RATE_REDUCED_OLD
+        newTaxRate = vatConstant.TAX_RATE_REDUCED_NEW
+    }
+
+    let taxRateDraftList =  validGermanTaxRateList.filter(rate => rate.amount === oldTaxRate)
+    if (taxRateDraftList.length>=1) {
```

Comment:
These checks should be done in a caller method and here you just process the tax rate (like the method name is saying).

Reply:
https://github.com/commercetools/commercetools-cli-scripts/pull/1/commits/79331752e59c927e5ee2748a164156f3e759c113